pyproject.toml

```diff
@@ -41,7 +41,9 @@ boto3 = "1.28.84"
 werkzeug = "2.3.7"
 itsdangerous = "2.1.2"
 
-certifi = "^2023.7.22" # Pinned for security issue: https://github.com/cds-snc/notification-utils/security/dependabot/22
+# Pinned dependencies
+certifi = "^2023.7.22" # pinned for security reasons: https://github.com/cds-snc/notification-utils/security/dependabot/22
+cryptography = "^41.0.2" # pinned for security reasons: https://github.com/cds-snc/notification-utils/security/dependabot/21
```

Comment:
Looks good - but it also looks like we currently have other issues that want it to be pinned to 41.0.4 or above.  Will we do that in a future PR then?

Reply:
Unless I am mistaken, this [should allow](https://python-poetry.org/docs/dependency-specification/#caret-requirements) versions `>=41.0.2 < 42.0.0`. If this isn't the case then yes, we'll have to bump this one again when we get to issues requiring `41.0.4`.

Ideally we wouldn't have to do this at all but as it's a transitive dep we're at the mercy of dep authors bumping their `cryptography` versions internally so we don't have to handle this manually.